nilearn/plotting/surface/_utils.py

```diff
@@ -109,17 +113,18 @@ def check_surface_plotting_inputs(
 ):
     """Check inputs for surface plotting.
 
-    Where possible this will 'convert' the inputs
-    if SurfaceImage or PolyMesh objects are passed
-    to be able to give them to the surface plotting functions.
+    Where possible this will 'convert' the inputs if
+    :obj:`~nilearn.surface.SurfaceImage` or :obj:`~nilearn.surface.PolyMesh`
+    objects are passed to be able to give them to the surface plotting
+    functions.
 
     Returns
     -------
-    surf_map : numpy.ndarray
+    surf_map : :obj:`numpy.ndarray`
 
-    surf_mesh : numpy.ndarray
+    surf_mesh : :obj:`numpy.ndarray`
 
-    bg_map : str | pathlib.Path | numpy.ndarray | None
+    bg_map : :obj:`str` | :obj:`pathlib.Path` | :obj:`numpy.ndarray` | None
 
     """
     if surf_mesh is None and surf_map is None:
```

Comment:
How does it sound to call `sanitize_hemi...` here, and return the final `hemi`  with other parameters?

Reply:
yes that'd be fine but we do not hemi=None anywhere then sanitize_hemi should now just be function that warns users that hemi is irrelevant if they have passed objects where hemi is not relevant (when they pass directly a mesh and an array for the data, instead of a polymesh + polydata or surface image...)